sdk/communication/azure-communication-networktraversal/src/main/java/com/azure/communication/networktraversal/CommunicationRelayClientBuilder.java

```diff
@@ -0,0 +1,303 @@
+import com.azure.core.annotation.ServiceClientBuilder;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.AzureKeyCredential;
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpPipelineBuilder;
+import com.azure.core.http.policy.BearerTokenAuthenticationPolicy;
+import com.azure.core.http.policy.CookiePolicy;
+import com.azure.core.http.policy.HttpLogOptions;
+import com.azure.core.http.policy.HttpLoggingPolicy;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.http.policy.RequestIdPolicy;
+import com.azure.core.http.policy.RetryPolicy;
+import com.azure.core.http.policy.UserAgentPolicy;
+import com.azure.core.util.ClientOptions;
+import com.azure.core.util.Configuration;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * CommunicationRelayClientBuilder that creates CommunicationRelayAsyncClient and CommunicationRelayClient.
+ */
+@ServiceClientBuilder(serviceClients = {CommunicationRelayClient.class, CommunicationRelayAsyncClient.class})
+public final class CommunicationRelayClientBuilder {
+    private static final String SDK_NAME = "name";
+    private static final String SDK_VERSION = "version";
+
+    private static final String COMMUNICATION_IDENTITY_PROPERTIES =
+        "azure-communication-networktravesal.properties";
+
+    private final ClientLogger logger = new ClientLogger(CommunicationRelayClientBuilder.class);
+    private String endpoint;
+    private AzureKeyCredential azureKeyCredential;
+    private TokenCredential tokenCredential;
+    private HttpClient httpClient;
+    private HttpLogOptions httpLogOptions = new HttpLogOptions();
+    private HttpPipeline pipeline;
+    private RetryPolicy retryPolicy;
+    private Configuration configuration;
+    private ClientOptions clientOptions;
+    private final Map<String, String> properties = CoreUtils.getProperties(COMMUNICATION_IDENTITY_PROPERTIES);
+    private final List<HttpPipelinePolicy> customPolicies = new ArrayList<HttpPipelinePolicy>();
+
+    /**
+     * Set endpoint of the service
+     *
+     * @param endpoint url of the service
+     * @return CommunicationRelayClientBuilder
+     */
+    public CommunicationRelayClientBuilder endpoint(String endpoint) {
+        this.endpoint = Objects.requireNonNull(endpoint, "'endpoint' cannot be null.");
+        return this;
+    }
+
+    /**
+     * Set endpoint of the service
+     *
+     * @param pipeline HttpPipeline to use, if a pipeline is not
+     * supplied, the credential and httpClient fields must be set
+     * @return CommunicationRelayClientBuilder
+     */
+    public CommunicationRelayClientBuilder pipeline(HttpPipeline pipeline) {
+        this.pipeline = Objects.requireNonNull(pipeline, "'pipeline' cannot be null.");
```

Comment:
These builder methods should allow setting `null` value to unset previously set values. The validation should happen in `build*Client()` methods

Reply:
Ohh, I see. I did it just as it is in communication-identity. Do you know if there's a sample I can look at? @srnagar

Reply:
In chat for some they're doing it as you say and makes a lot of sense in those cases which are values that can be null. They just check, and don't use it if it's null.  

In this case, if I add the checks later I will need to throw there, which in the end will be the same having the check on one place or the other. If we would need to reset, we should assign to something else the value and not be able to set to null.

Reply:
In general, we should try to validate when calling the `build` method instead of doing the validation on individual setter methods. That's the recommendation anyway. If we need to reset to default, there's no way for the user to know what the default is for a httpPipeline. So, setting to `null` would be a way for the build method to create the default pipeline as it usually would.